### ISSUE TITLE
UnexpectedValueException: There is no existing directory "storage/logs"

### ISSUE THREAD

**Issue by hopkins385** [serversideup/docker-php#95]
### Affected Docker Images

```
serversideup/php:8.1-fpm-nginx
```
docker-compose.dev.yml
```
version: '3.8'
services:   
  laravel:
    image: serversideup/php:8.1-fpm-nginx
    container_name: laravel
    restart: unless-stopped
    environment:
      PUID: "501"
      PGID: "20"
      PRODUCTION: "false"
      SSL_MODE: "off"
      AUTORUN_LARAVEL_MIGRATION: "true"
      AUTORUN_LARAVEL_STORAGE_LINK: "true"
      PHP_POOL_NAME: "laravel-app_php"
    ports:
      - 80:80
      - 443:443
    command: 
    # run larvel scheduler and horizon in the background
    - su
    - webuser
    - -c
    - |
      php artisan horizon &
      php artisan schedule:work
    volumes:
      - .:/var/www/html/:cached
    depends_on:
      - "mariadb"
      - "redis"
    networks:
      - laravel
  ```

### Current Behavior

Hi guys, it seems as if the the storage/logs folder cannot be found which is a bit weird, because on my external server everything is working fine. Just on my home-laptop not. 

I tried to play around with the UID and GID and even changed the permission of the folder "storage" to 777. 
Like so `sudo chmod -R 777 storage` without success.  

UID and GID are set as per ìd` output in zsh on mac.

I even tried to run the container in privileged, without success. 

Am I missing something?

Here is the log:

```
laravel          | 
laravel          | User uid:    501
laravel          | User gid:    20
laravel          | -------------------------------------
laravel          | 
laravel          | 🔓 SSL_MODE has been DISABLED, setting the web server to work in HTTP only...
laravel          | 🏃‍♂️ Checking for Laravel automations...
laravel          | 🚀 Running migrations...
laravel          | 
laravel          |    UnexpectedValueException 
laravel          | 
laravel          |   There is no existing directory at "/Users/sven/var/www/dash/storage/logs" and it could not be created: Permission denied
laravel          | 
laravel          |   at /var/www/html/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php:216
laravel          |     212▕             set_error_handler([$this, 'customErrorHandler']);
laravel          |     213▕             $status = mkdir($dir, 0777, true);
laravel          |     214▕             restore_error_handler();
laravel          |     215▕             if (false === $status && !is_dir($dir) && strpos((string) $this->errorMessage, 'File exists') === false) {
laravel          |   ➜ 216▕                 throw new \UnexpectedValueException(sprintf('There is no existing directory at "%s" and it could not be created: '.$this->errorMessage, $dir));
laravel          |     217▕             }
laravel          |     218▕         }
laravel          |     219▕         $this->dirCreated = true;
laravel          |     220▕     }
laravel          | 
laravel          |       +11 vendor frames 
laravel          |   12  [internal]:0
laravel          |       Illuminate\Foundation\Bootstrap\HandleExceptions::Illuminate\Foundation\Bootstrap\{closure}()
```

### Expected Behavior

The existing directory should be found.

### Steps To Reproduce

1. Download docker for mac
2. Create new laravel project
3. Create docker compose file as I have mentioned, and extend that file with redis and mariadb in standard config
4. run `docker compose -f docker-compose.dev.yml up -d  `

### Host Operating System

macOS Ventura 13.0.1, Apple Silicon

### Docker Version

```markdown
Client:
 Cloud integration: v1.0.29
 Version:           20.10.21
 API version:       1.41
 Go version:        go1.18.7
 Built:             Tue Oct 25 18:01:18 2022
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.14.0 (91374)
 Engine:
  Version:          20.10.21
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.7
  Built:            Tue Oct 25 17:59:41 2022
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.9
 runc:
  Version:          1.1.4
 docker-init:
  Version:          0.19.0
```

**Comment by hopkins385** [comment]
Oookay ... I found the solution ... this is what happened. 

I perviously used Valet on my local machine and cached a `.env.local` env-file via `php artisan config:cache --env=local` for the local development. 

Then I shut down all local services via `valet stop` renamed the env.local to '.env' and changed the content of that so that it can be used with docker.

Then I fired up the docker compose instance and was confronted with the described issue.

After hours of figuring out what could be the root issue, I reverted everything back and! changed the content of the .env file back to being able to work on localhost (127.0.0.1). I also did a `php artisan optimize:clear`.

Then I though .. ok .. lets give it another try ... backed up the .env to .env.local and made again all the needed changed in the .env to be able to run on docker. .. and voila ... everything is up running now 👍 

Sorry for the false report/confusion, but maybe an helpful experience if anyone else runs into similar issues when moving from classic valet to docker. 

Issue can be closed.
